rest/routes/user.py

```diff
@@ -0,0 +1,139 @@
+        data['invalid_password'] = password
+
+    if len(data) > 0:
+        response = Response(
+            response=json.dumps(data),
+            status=400,
+            mimetype='application/json'
+        )
+        return response
+
+    check_user = User.query.filter_by(email=email).first()
+
+    if check_user:
+        response = Response(
+            response=json.dumps({'email_taken': email}),
+            status=409,
+            mimetype='application/json'
+        )
+        return response
+
+    new_user = User(email, name, password, False)
+    db.session.add(new_user)
+
+    if editor_request:
+        new_editor_request = EditorRequest(email, name)
+        db.session.add(new_editor_request)
+
+    db.session.commit()
+
+    response = Response(
+        response=json.dumps({}),
+        status=200,
+        mimetype='application/json'
+    )
+    return response
+
+
+@user_url.route('/auth', methods=['POST'])
+def login():
+    email = request.form.get('email')
+    password = request.form.get('password')
+
+    user = User.query.filter_by(email=email).first()
+
+    if user and user.password == password:
+        if user.email == 'admin@admin.agh.edu.pl':
+            role = 'admin'
+        elif user.is_editor:
+            role = 'editor'
+        else:
+            role = 'user'
+
+        data = {'auth': True, 'role': role}
+        status = 200
+    else:
+        data = {'auth': False}
+        status = 400
+
+    response = Response(
+        response=json.dumps(data),
+        status=status,
+        mimetype='application/json'
+    )
+    if status == 200:
+        response.set_cookie('email', value=email)
+
+    return response
+
+
+@user_url.route('/editorRequests', methods=['GET', 'POST'])
+def admin_editor_requests():
+    if request.method == 'POST':
+        email = request.form.get('email')
+        name = request.form.get('name')
+
+        check_user = User.query.filter_by(email=email).first()
+        check_editor_request = EditorRequest.query.filter_by(user_email=email).first()
+
+        if not check_user or not check_editor_request or \
+                check_user.name != name or check_editor_request.name != name:
+            status = 400
+            data = {'data': 'invalid'}
+        else:
+            check_user.is_editor = True
+            db.session.delete(check_editor_request)
+            db.session.commit()
+
+            status = 200
+            data = {'data': 'valid', 'is_editor': check_user.is_editor}
+
+        return Response(
+            response=json.dumps(data),
+            status=status,
+            mimetype='application/json'
+        )
+
+    if 'email' not in request.cookies:
+        status = 401
+        data = {'cookie': False}
+    else:
+        email = request.cookies.get('email')
+        user = User.query.filter_by(email=email).first()
+
+        if not user or not email == 'admin@admin.agh.edu.pl':
```

Comment:
do we have to hardcode that?

Reply:
We need to hardcode some form of admin email style or put in is_admin field in database. I could change it to anything ending with "@admin.agh.edu.pl" is an admin. I don't see any other way